From 5a1841cee4e1301e652816b03be565355fa2a812 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@freescale.com>
Date: Fri, 30 Oct 2015 17:31:31 +0800
Subject: [PATCH 0751/5242] MLK-11800 dts: imx7d-sdb: correct flexcan stby
 gpio pin

commit  6d86b8c57d185320a736394372db47977e12e90f from
https://source.codeaurora.org/external/imx/linux-imx.git

According to schematic, the pad used for CAN STBY signal
should be EPDC_DATA14, not GPIO1_IO07. So correct it.
And due to pin is conflict with epdc, so we also update
the imx7d-sdb-epdc.dts.

Reported-by: Peng Fan <Peng.Fan@freescale.com>
Signed-off-by: Dong Aisheng <aisheng.dong@freescale.com>

Conflicts:
	arch/arm/boot/dts/imx7d-sdb.dts
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-sdb-epdc.dts |    4 ++++
 arch/arm/boot/dts/imx7d-sdb.dts      |    4 ++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-sdb-epdc.dts b/arch/arm/boot/dts/imx7d-sdb-epdc.dts
index 2cc8045..2bad80a 100644
--- a/arch/arm/boot/dts/imx7d-sdb-epdc.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-epdc.dts
@@ -20,6 +20,10 @@
 	status = "disabled";
 };
 
+&flexcan2 {
+	status = "disabled";
+};
+
 &max17135 {
         status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 24f4a44..c002d67 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -77,7 +77,7 @@
 		regulator-name = "can2-3v3";
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
-		gpio = <&gpio1 7 GPIO_ACTIVE_LOW>;
+		gpio = <&gpio2 13 GPIO_ACTIVE_LOW>;
 	};
 
 	reg_vref_1v8: regulator-vref-1v8 {
@@ -618,7 +618,6 @@
 		pinctrl_hog_2: hoggrp-2 {
 			fsl,pins = <
 				MX7D_PAD_LPSR_GPIO1_IO05__GPIO1_IO5	  0x14
-				MX7D_PAD_LPSR_GPIO1_IO07__GPIO1_IO7    0x59
 			>;
 		};
 
@@ -817,6 +816,7 @@
 			fsl,pins = <
 				MX7D_PAD_UART3_CTS_B__GPIO4_IO7		0x14
 				MX7D_PAD_ECSPI2_SS0__GPIO4_IO23		0x80000000  /* bt reg on */
+				MX7D_PAD_EPDC_DATA14__GPIO2_IO14 	0x59   /* CAN_STBY */
 			>;
 		};
 
-- 
1.7.9.5

